observability/opentelemetry/v2/client/otel_options.go

```diff
@@ -37,6 +38,15 @@ func WithSpanNameFormatter(nameFormatter func(cloudevents.Event) string) OTelObs
 	}
 }
 
+// WithTracerProvider sets the tracer provider to use for creating spans.
+func WithTracerProvider(tracerProvider trace.TracerProvider) OTelObservabilityServiceOption {
+	return func(os *OTelObservabilityService) {
+		if tracerProvider != nil {
```

Comment:
Why not let them set it to 'nil' if they want?

Reply:
To prevent, panic.
If someone want to disable trace, we can use noop.TracerProvider.

https://github.com/open-telemetry/opentelemetry-go/blob/trace/v1.38.0/trace/noop/noop.go#L36